docroot/modules/humsci/hs_migrate/src/Form/HsMigrateImportForm.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Drupal\hs_migrate\Form;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Datetime\DateFormatterInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\KeyValueStore\KeyValueFactoryInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\hs_migrate\HsMigrateBatchExecutable;
+use Drupal\migrate\Plugin\MigrationPluginManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\migrate\MigrateMessage;
+use Drupal\migrate\Plugin\MigrationInterface;
+
+/**
+ * Class HsMigrateImportForm.
+ *
+ * @package Drupal\hs_migrate\Form
+ */
+class HsMigrateImportForm extends FormBase {
+
+  /**
+   * Array of migration plugin objects.
+   *
+   * @var \Drupal\migrate\Plugin\MigrationInterface[]
+   */
+  protected $migrations;
+
+  /**
+   * Date formatter service.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatterInterface
+   */
+  protected $dateFormatter;
+
+  /**
+   * Key Value collection of last migrations.
+   *
+   * @var \Drupal\Core\KeyValueStore\KeyValueStoreInterface
+   */
+  protected $lastMigrations;
+
+  /**
+   * Migration plugin manager service.
+   *
+   * @var \Drupal\migrate\Plugin\MigrationPluginManagerInterface
+   */
+  protected $migrationManager;
+
+  /**
+   * Current user account.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $account;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.migration'),
+      $container->get('date.formatter'),
+      $container->get('keyvalue'),
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(MigrationPluginManagerInterface $migrations_manager, DateFormatterInterface $date_formatter, KeyValueFactoryInterface $key_value, AccountProxyInterface $account) {
+    $this->dateFormatter = $date_formatter;
+    $this->lastMigrations = $key_value->get('migrate_last_imported');
+    $this->migrationManager = $migrations_manager;
+    $this->account = $account;
+
+    $this->migrations = $this->migrationManager->createInstances([]);
+
+    // No need to show migrations that are dependencies. They will get executed
+    // when their dependent migration is executed.
+    foreach ($this->migrations as $migration) {
+      foreach ($migration->getMigrationDependencies()['required'] as $dependency) {
+        // Set the dependency migrations to idle so that the primary migration
+        // will still execute.
+        $this->migrations[$dependency]->interruptMigration(MigrationInterface::RESULT_STOPPED);
```

Comment:
Is this for when migrations are running and you are in the UI? Do we really want to halt them? What would the outcome be if a migration was halted mid-run and the user did not re-run it through the form?

Reply:
yeah this is via the UI. it forces the dependent migrations run be executed. if a migration is interrupted during import, it require manual manipulation to reset it's status so that it can run again. It shouldn't be an issue here because it'll run through all the items again, skipping those that have already been imported.

Reply:
My question is do you really want to do this by simply visiting the form? Wouldn't you want this to be a deliberate action?

Reply:
ahh. i hear ya there. :+1: yeah i think we could do this part during the actual execution.

Reply:
I moved the migration reset into the batch preparation. so it will only reset the necessary migrations as well.